### PR TITLE
ref(actix): Migrate the EnvelopeManager actor

### PR DESCRIPTION
**Pull request by jan-auer**
Changes the `EnvelopeManager` to run as service on the main tokio
runtime:

- `Capture` and `GetCapturedEnvelope` are moved into a dedicated service
  called `TestStore`. This service serializes its messages and mutates a
  map of stored envelopes for testing.
- The remaining messages are handled by `EnvelopeManager` in parallel.
  It continues to await the response for sending Envelopes and handles
  errors.
- The `Store` service is now injected at construction so that the
  `EnvelopeManager` no longer needs to hold a tokio runtime internally.
  That runtime is now created in the overall service constructor and can
  be removed in the future.
- The `SendMetrics` message still has an asynchronous response,
  returning the buckets on error. This can be changed in a follow-up.

#skip-changelog

### CODE REVIEW

**Comment by jan-auer** [comment]
Tests are currently failing because the `EnvelopeContext` unconditionally invokes `Capture`.

File: relay-server/src/actors/envelopes.rs [hunk 1]
```diff
@@ -148,79 +141,135 @@ impl UpstreamRequest for SendEnvelope {
...
+    /// The key of the logical partition to send the metrics to.
+    pub partition_key: Option<u64>,
+}
+
+/// Dispatch service for generating and submitting Envelopes.
+#[derive(Debug)]
+pub enum EnvelopeManager {
+    SubmitEnvelope(Box<SubmitEnvelope>),
+    SendClientReports(SendClientReports),
+    SendMetrics(SendMetrics, Sender<Result<(), Vec<Bucket>>>),
```
Comment (jjbayer) [inline relay-server/src/actors/envelopes.rs:179]: nit: Would be nice to align these names, so either all `Submit*` or all `Send*`.

File: relay-server/src/service.rs [hunk 2]
```diff
@@ -142,10 +146,13 @@ impl ServiceState {
         let system = System::current();
         let registry = system.registry();
 
+        let main_runtime = utils::tokio_runtime_with_actix();
+        let outcome_runtime = utils::tokio_runtime_with_actix();
+        let mut _store_runtime = None;
```
Comment (jjbayer) [inline relay-server/src/service.rs:151]: Why the leading underscore?

**Review by jjbayer — APPROVED** [review]
Not an expert on the new tokio-based services, but this looks good to me.

File: relay-server/src/actors/envelopes.rs (hunk 1 above)
Comment (jan-auer) [reply relay-server/src/actors/envelopes.rs:179]: Sounds good, will add this in a follow-up.

File: relay-server/src/service.rs (hunk 2 above)
Comment (jan-auer) [reply relay-server/src/service.rs:151]: It silences a warning since the runtime is unused if the `processing` feature is disabled.

**Pull request merged**